src/main/java/org/javarosa/core/model/condition/IFunctionHandler.java

```diff
@@ -16,37 +16,45 @@
 
 package org.javarosa.core.model.condition;
 
+import org.jetbrains.annotations.NotNull;
+
 import java.util.List;
 
 public interface IFunctionHandler {
+
     /**
      * @return The name of function being handled
      */
-    String getName();
+    @NotNull String getName();
 
     /**
      * @return List of allowed prototypes for this function. Each prototype is
-     *         an array of Class, corresponding to the types of the expected
-     *         arguments. The first matching prototype is used.
+     * an array of Class, corresponding to the types of the expected
+     * arguments. The first matching prototype is used.
      */
-   List<Class[]> getPrototypes();
+    @NotNull List<@NotNull Class<?> @NotNull[]> getPrototypes();
 
     /**
      * @return true if this handler should be fed the raw argument list if no
-     *         prototype matches it
+     * prototype matches it
      */
-    boolean rawArgs();
+    default boolean rawArgs() {
+        return false;
+    }
 
     /**
      * @return true if the result of this handler depends on some dynamic data
-     *         source, and the expression cannot be pre-computed before the
-     *         question is reached (un-supported)
-     *
+     * source, and the expression cannot be pre-computed before the
+     * question is reached
+     * @deprecated un-supported
      */
-    boolean realTime();
+    @Deprecated
+    default boolean realTime() {
```

Comment:
I couldn't find a time this was ever used. It looks to me that came with a [massive contribution](https://github.com/getodk/javarosa/commit/65101db09a6cf0cf8f293b1502e783eda74143d8) that it probably got lost in.